Dockerfile

```diff
@@ -59,15 +71,19 @@ FROM no-spark AS with-spark-optional-dag
 
 # Install Java
 RUN apt-get update \
-  && apt-get install -t jessie-backports --no-install-recommends -y openjdk-8-jre-headless \
-  && rm -rf /var/lib/apt/lists/*
+    && apt-get install -t jessie-backports --no-install-recommends -y openjdk-8-jre-headless \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG java_home=/usr/lib/jvm/java-8-openjdk-amd64
+ENV JAVA_HOME=${java_home}
```

Comment:
Build args should not be permitted as the installation is done in this Dockerfile.

Also it is possible to programmatically set the JAVA_HOME with the command `type -p javac | xargs readlink -f | xargs dirname | xargs dirname`

Reply:
Thanks for the helpful command, will try it

Reply:
As discussed, using a runtime command to set the environment var might be flakey (https://forums.docker.com/t/how-do-i-send-runs-output-to-env-in-dockerfile/16106), so will stick to setting it statically, but straight to `ENV JAVA_HOME` instead.

Reply:
Yes you are indeed correct, I overlooked that this can only be executed with `RUN` not `ENV`.